backend/services/implementations/userService.ts

```diff
@@ -16,6 +17,41 @@ const getMongoUserByAuthId = async (authId: string): Promise<User> => {
   return user;
 };
 
+const getDateTime = (date: string): string =>
+  `${Math.floor(new Date(date).getTime() / 1000).toString(16)}0000000000000000`;
+
+interface MatchObject {
+  $match: {
+    _id: {
+      [key: string]: Types.ObjectId;
+    };
+  };
+}
```

Comment:
ehh I'm not sure about best preference, I kept it here because it's private to this file (they could be anonymous to the same effect, this makes it a tad more readable)

Reply:
updated name atm is `LimitByDateQuery`

Reply:
Yeah I think you're right after sleeping on it. There are other files where types are defined inline with the functions that use them.